### PR TITLE
Optimized SortFlags to minimize Splitter usage

### PR DESCRIPTION
**Pull request by MfgSolid**
In the previous State the SplitterGenerator created arrays that still had the potential use less Splitters.

Original Output for 5-Way Split with Ratios 180, 60, 30, 30, 30:
                0
       0               0
   0       0       0       X
 0   0   0   1   1   0   .   .
4 2 2 5 3 1 . . . . 1 X . . . .

Optimized Output:
               0
       0               0
   1       0       0       X
 .   .   1   2   0   0   .   .
. . . . . . . . 3 4 5 X . . . .

This was achieved by sorting the values in SortFlags into odd and even lists by the occurences of output on the final level instead of the index of the given output.

### CODE REVIEW

**Comment by MfgSolid** [comment]
Better Formatting for Examples

grafik

**Comment by icorbrey** [comment]
Hey @MfgSolid! Thank you for your PR, this project is so old that I wasn't anticipating receiving any. Would you mind explaining your changes from a technical standpoint? I've never worked with histograms like this and I'm curious how they are intended to operate

**Comment by MfgSolid** [comment]
Uh, after some time thinking about it the actual data structure might be more of a frequency table; The histogram being a Visualization for that in some cases.


**TL;DR**
In your SortfFlags you had the right idea, but instead of classifying the amount of outputs odd or even, you classified the output index.

**Long Version**
What I did was effectively counting the amount of final outputs and classifiying them as odd or even. Then from left to right I inserted all as even classified outputs, then odd, then the return outputs (X)

So in this example I have the following final layer:
4 2 2 5 3 1 1 1 1 1 1 X X X X X

Your Condense algorithm then merges splitters if they have the same output (Examples in Brackets)
4 2 2 5 3 1 [1 1] [1 1] 1 X X X X X

This is sometimes inopportune as with [4 2] in the beginning as they mismatch, even though there is an even number of 2s in the list. All I did was rearrange the outputs, so that the amount of splitters reduced is maximized. For this I first put even outputs in the final layer, as they can be completely condensed. The following odd layers can also be condensed, if they have at least 3 outputs. This is further enhanced by 2 adjacent odd groups. The rest is filled with Xs.

E.g.:
[11][11][11][22]345X[XX][XX]

I hope this wasn't too confusing
